examples/misc/lib/language_tour/classes/point_alt.dart

```diff
@@ -4,18 +4,24 @@
 /// - A constructor initializing fields in the body "the long way"
 /// - A named constructor with initializers, and a print statement in the body.
 ///
-// #docregion constructor-long-way
+// #docregion constructor-long-way, idiomatic-constructor
 class Point {
   double x = 0;
   double y = 0;
+  // #enddocregion constructor-long-way
+
+  Point(this.x, this.y);
+  // #enddocregion idiomatic-constructor
 
+  /* 
+  // #docregion constructor-long-way
+  
   Point(double x, double y) {
-    // See initializing formal parameters for a better way
-    // to initialize instance variables.
     this.x = x;
```

Comment:
While we are here, in Dart you really shouldn't be assigning fields inside of constructor bodies ever, so it does feel weird for me to highlight this as an example. Especially with null safety, this often would force you into a `late` or nullable field.

This might be what some people are used to seeing in other languages, but you really don't ever want to do this in Dart. If you need to do some logic that you can't fit into the initializer list, you should create a factory constructor with that logic and then pass the computed values from there into a normal constructor.

What if we changed this example to use an explicit initializer list? That is the more canonical "long way" constructor.

Reply:
@jakemac53 I removed the longer example all together because I didn't think it was necessary to show a wrong way at all. 

But, if you think a lot of users default to writing out explicit initializer lists without realizing they don't have to, then I can add it back in as the longer "Instead of..." part of the example. What would the syntax for that look like though? Same as the [Initializer lists](https://dart.dev/language/constructors#initializer-list) section?

> In order to perform some logic that cannot be expressed in the initializer list, it is better to create a factory constructor (or static method) with that logic and then pass the computed values to a normal constructor.

I added that note to the end of the initializing formal parameters section, to point to factory constructors and static methods. Seems like it'd be helpful to know if you're reading the page from top to bottom and the first thing you see is the simplistic initializing formal parameters example, but let me know if that doesn't make sense, thanks!

Reply:
I definitely think it makes a lot of sense for the simplistic example to be at the top. The transition between that and the next section is a bit awkward now though - maybe just add a comment on the first constructor, saying the next section will describe what it is doing?.